docs/markdown/star_formation.md

```diff
@@ -0,0 +1,33 @@
+Adding star particle to every Jeans violating cell can lead to a large number of small mass stars which can be cumbersome. To alleviate this issue, we control the star formation rate through $\epsilon_{\rm ff}$ and $\epsilon_{*}$. $\epsilon_{\rm ff}$ is the efficiency per freefall time given by $=\dot{M}_* \cdot t_{\rm ff} / M_{\rm cell} \cdot dt $, while $\epsilon_{*}$ is the fraction of cell mass that gets converted into stars. 
+
+There are two equivalent ways of converting a certain mass from cell into stars. 
+
+In the first, we assume that every cell that is Jeans unstable definitely forms a star. In this scenario, $M_{*} = \epsilon_{\rm ff} \cdot dt \cdot M/t_{\rm ff}$ is the mass that gets converted into stars from a cell mass of M. This will lead to the issue mentioned earlier, that is of a large number of stars being created in a galaxy-scale simulation.
+
+We circumvent this by associating a probability P to the process of spawning a star in the cell. A choice for P $=\epsilon_{\rm ff} \cdot dt/\epsilon_{*}$, which is ratio of the expected from the star formation rate in the cell over the timestep dt ($\epsilon_{\rm ff} dt/t_{\rm ff}$) to the fraction of mass available for star formation ($\epsilon_{\rm ff} M$). It should be noted that this is the probability for $\epsilon_{\rm ff}$ averaged over the timestep rather than for the instantaneous star formation rate, which should be integrated over the timestep dt. 
+
+Both the descriptions of P, from assuming a star formation rate averaged over timestep or an instantaneous star formation rate, lead to identical expression of P if $dt<t_{\rm ff}$. However if we are in the regime of $dt>t_{\rm ff}$ neither prescriptions of P offer the correct answer because they rely on conditions at the beginning of the timestep. 
+
+
+## Estimating number and type of stars
+
+Once we establish that a star will form in a cell, we need to find out the stellar population looks like. In our implementation, the population comprises at least particle representing low mass star ($M<8M_{\odot}$) population and a random number of particles, each represeting a high mass star.
+
+The number of high mass stars is determined by the initial mass function, taken to be Chabrier03. The IMF is represented by a log normal for $M<M_{\odot}$ and a power law with a slope of $2.35$ beyond that. From the IMF, we can estimate the fraction of high mass stars, $f_{*,\rm{high}}$, and the average mass of high mass stars, $<m>_{*,\rm{high}}$. The expectation value of the number of high mass stars, \textit{num_high_mass_stars}, then is the ratio of the total mass in high mass stars and $<m>_{*,\rm{high}}$. 
+
+The total number of star particles to be spwaned in a cell is $1+$ Poission distribution with an expectation value of \textit{num_high_mass_stars}. A fraction $1-f_{*,\rm{high}}$ goes into the particle representing low mass star while the mass of the high mass star particles is randomly drawn from the high mass part of the IMF.
```

Comment:
There is a LaTeX formatting issue in this paragraph:
<img width="750" alt="Screenshot 2025-04-16 at 8 23 28 PM" src="https://github.com/user-attachments/assets/d8db66d8-c573-45c2-a150-b7e8a8c6cdaf" />